### ISSUE TITLE
Audible deterioration when rendering clip stretching

### ISSUE THREAD

**Issue by dozzzzer** [audacity/audacity#5390]
### Bug description

As the title says, when using the context menu command "Render Clip Stretching" on non-synthetic tones, it produces an oddly sounding result, as if channels are out of phase, causing audible changes to the impact of the sound as well as its stereo image. 

It's also worth mentioning that mixing-and-rendering via `Tracks > Mix > Mix and Render` doesn't cause such an issue.

### Steps to reproduce

1. Download & open[ this project ](https://www.dropbox.com/scl/fi/8q5nxf98biy4pjonneai9/Render-Stretching-bug.aup3?rlkey=o9arietc3qccdu7nyzeqwfagz&dl=0)
2. Make sure looping is enabled and hit Play
3. By pressing the **Shift + Solo** button, compare all three tracks where the 1st is the original drum loop shrunk by 6%, the second one is the result of the `Tracks > Mix > Mix and Render` command, and the third one is the duplicate of the 1st track with the "Render Clip Stretching" context menu command applied to it (headphones are highly recommended)

@Paul-Licameli I suggest this testing method too:  Mix-and-render to a new track, then apply Invert to that track, and then play all tracks.  Listen, expecting silence.  Also watch the playback meter (maybe using the -145 dB setting in Interface preferences to see even small, inaudible non-zero signals).

### Expected behavior

All three sound equal

### Actual behavior

The 1st and the 2nd one sound identical;
The 3rd track sounds somewhat different from the previous two, introducing an altered stereo image and less impactful and "tight", "washed out" sounding kick and snare drums.

### Audacity Version

[current master / alpha / nightly build
](https://github.com/audacity/audacity/actions/runs/6546774311)

### Operating system

All

### Additional context

It seems that this behavior has been introduced since the very appearance of this command in the clip context menu, so it's not a regression.

QA (to be added to PR when there is one):
- [ ] Doesn't re-introduce #5313
- [ ] If a clip has no trimming, rendering is with RMB or with mix-and-render command yields bit-exact result
- [ ] Whether stretch-rendering a clip and then apply and effect **or** directly applying the effect yields bit-exact result
- [ ] All the above was done in mono and stereo

**Comment by saintmatthieu** [comment]
Thanks @dozzzzer !
The waveforms are indeed visibly different depending on the chosen rendering method. Looking into this.

**Comment by saintmatthieu** [comment]
I know what the problem is, and it's an embarrassing one ...

TLDR:  Stretch-rendering through mix-and-render or export are joint-stereo, and through the RMB, dual mono.

Detail: Mix-and-Render passes instances of the `WideClip` implementation of the `ClipInterface` to the stretching algorithm.
That implementation provides joint stereo audio by combining the mono audio of the left and right clip.
The RMB renders the stretching on the two clips one after the other, i.e. in dual mono.

I don't know yet how to fix this.

**Comment by dozzzzer** [comment]
Yes, it indeed affects joining clips with different stretch ratios (in which case rendering is applied) as well as destructive effects application.

**Comment by DavidBailes** [comment]
Using https://www.voxengo.com/product/correlometer/ to detect phase issues, this is a screenshot for correlometers for tracks 2 and 3, at some arbitrary playback time:
![image](https://github.com/audacity/audacity/assets/11887021/c5b19f17-3ace-4056-a419-121b6ef62948)
